### PR TITLE
Retry on closed connection

### PR DESCRIPTION
**Pull request by eminugurkenar**
We upload large amount of data to s3 and after upgrading version 1.0.0 ,  we experienced failures for large uploads , randomly we get errors like below.

`RequestError: send request failed
caused by: Put "https://testing.s3.amazonaws.com/xxxxx&partNumber=n&uploadId=id": write tcp 10.xx.xx.xx:41654->52.216.8.115:443: use of closed network connection`

After digging AWS SDK `ShouldRetry` method, we noticed that for some cases it does return `false` but for some cases it returns `true` even if base error says `use of closed network connection` in both cases, you can see below

`SerializationError: failed to unmarshal error message
        status code: 500, request id: IDDDDDDD, host id: 3231sdsadasdasdI=
caused by: UnmarshalError: failed to unmarshal error message
caused by: read tcp 10.xx.xx.xx:39860->52.216.166.67:443: use of closed network connection: true`

`RequestError: send request failed
caused by: Put "https://testing.s3.amazonaws.com/xxxx?partNumber=n&uploadId=id--": write tcp 10.xx.xx.xx:41654->52.216.8.115:443: use of closed network connection: false`

So we implemented this patch to retry on any error that contains `use of closed network connection` ,  as golang does not export this error `https://github.com/golang/go/issues/4373` , we had to check error msg.

### CODE REVIEW

File: storage/s3.go [hunk 1]
```diff
@@ -627,6 +627,11 @@ func (c *customRetryer) ShouldRetry(req *request.Request) bool {
 	if errHasCode(req.Error, "InternalError") {
 		return true
 	}
+
+	if errContains(req.Error, "use of closed network connection") {
```
Comment (igungor) [inline storage/s3.go:631]: could you update S3Retry test?

File: storage/s3.go [hunk 2]
```diff
@@ -672,6 +677,22 @@ func errHasCode(err error, code string) bool {
...
+
+	var awsErr awserr.Error
+	if errors.As(err, &awsErr) {
+		if strings.Contains(awsErr.Error(), msg) {
+			return true
+		}
+	}
+
+	return false
+
```
Comment (igungor) [inline storage/s3.go:693]: remove trailing newline

File: storage/s3.go (hunk 1 above)
Comment (eminugurkenar) [reply storage/s3.go:631]: updated

File: storage/s3.go (hunk 2 above)
Comment (eminugurkenar) [reply storage/s3.go]: removed

File: storage/s3_test.go [hunk 3]
```diff
@@ -283,6 +283,10 @@ func TestS3Retry(t *testing.T) {
 			name: "RequestError",
 			err:  awserr.New(request.ErrCodeRequestError, "request error", nil),
 		},
+		{
+			name: "RequestError",
+			err:  awserr.New(request.ErrCodeRequestError, "use of closed network connection", nil),
```
Comment (igungor) [inline storage/s3_test.go:288]: could you use a unique test name please?

Comment (eminugurkenar) [reply storage/s3_test.go:288]: updated

**Pull request merged**